Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs

```diff
@@ -47,30 +47,63 @@ public override void OnNetworkSpawn()
 
             this.RegisterNetworkUpdate(updateStage: NetworkUpdateStage.Update);
             this.RegisterNetworkUpdate(updateStage: NetworkUpdateStage.FixedUpdate);
+
+            var cameraControl = Camera.main?.GetComponent<CameraControl>();
+            if (cameraControl != null)
+            {
+                cameraControl.SetTransform(transform);
+                SetCameraReference(Camera.main);
+            }
+            else
+            {
+                Debug.LogError("CameraControl not found on the Main Camera or Main Camera is missing.");
+            }
         }
 
         public override void OnNetworkDespawn()
         {
             base.OnNetworkDespawn();
-            if (m_AvatarInputs)
+
+            if (m_AvatarInputs != null)
             {
                 m_AvatarInputs.Jumped -= OnJumped;
             }
 
             this.UnregisterAllNetworkUpdates();
+
+            var cameraControl = Camera.main?.GetComponent<CameraControl>();
+            if (cameraControl != null)
+            {
+                cameraControl.SetTransform(null);
+            }
         }
 
-        void OnJumped()
+        private void OnJumped()
         {
             m_PhysicsPlayerController.SetJump(true);
         }
 
-        void OnTransformUpdate()
+        private void OnTransformUpdate()
         {
-            var movement = new Vector3(m_AvatarInputs.Move.x, 0, m_AvatarInputs.Move.y).normalized;
+            if (m_MainCamera != null)
+            {
+                Vector3 forward = m_MainCamera.transform.forward;
+                Vector3 right = m_MainCamera.transform.right;
 
-            m_PhysicsPlayerController.SetMovement(movement);
-            m_PhysicsPlayerController.SetSprint(m_AvatarInputs.Sprint);
+                forward.y = 0f;
+                right.y = 0f;
+                forward.Normalize();
+                right.Normalize();
+
+                Vector3 movement = forward * m_AvatarInputs.Move.y + right * m_AvatarInputs.Move.x;
+                m_PhysicsPlayerController.SetMovement(movement);
+                m_PhysicsPlayerController.SetSprint(m_AvatarInputs.Sprint);
+            }
+        }
+
+        internal static void SetCameraReference(Camera camera)
```

Comment:
Can remove internal static from here, or remove this method entirely if it's only invoked in this class.